src/dftbp/dftbplus/parser.F90

```diff
@@ -7624,4 +7632,55 @@ function parserVersionFromInputVersion(versionString, node) result(parserVersion
 
   end function parserVersionFromInputVersion
 
+
+  !> Get the environment variable describing the search path for DFTB+
+  subroutine getSKPath(path)
+    type(string), allocatable, intent(out) :: path(:)
+
+    character(len=*), parameter :: skEnv = "DFTBPLUS_PATH"
+    character(len=:), allocatable :: var
+
+    call getEnvVar(skEnv, var)
+
+    if (allocated(var)) then
+      allocate(path(1))
+      path(1) = var
+    else
+      allocate(path(0))
+    end if
+  end subroutine getSKPath
+
+
+  subroutine findFile(path, inName, outName)
+    type(string), intent(in) :: path(:)
+    character(len=*), intent(in) :: inName
+    character(len=:), allocatable :: outName
+
+    integer :: ip
+    logical :: exists
+
+    if (isAbsolutePath(inName)) then
+      outName = inName
+      return
+    end if
+
+    inquire(file=inName, exist=exists)
+    if (exists) then
+      outName = inName
+      return
+    end if
+
+    do ip = 1, size(path)
+      outName = char(path(ip))//"/"//inName
+      inquire(file=outName, exist=exists)
+      if (exists) exit
+
+      outName = char(path(ip))//"\"//inName
+      inquire(file=outName, exist=exists)
+      if (exists) exit
+    end do
+
+    if (.not.exists .and. allocated(outName)) deallocate(outName)
```

Comment:
But this will always deallocate the string we want to return. This branch should only be taken if the search was unsuccessful, *e.g.* we fall through the loop after exceeding all paths we can search